pyfar/plot/_utils.py

```diff
@@ -457,6 +457,34 @@ def _assert_and_match_data_to_side(data, signal, side):
     return data, frequencies, xlabel
 
 
+def _assert_and_match_spectrogram_to_side(spectrogram, frequencies, signal,
+                                          side):
+    """Adjust data and frequency vector for plotting as specified by side."""
+
+    if side == 'left':
+        mask = frequencies <= 0
+    elif side == 'right':
+        mask = frequencies >= 0
+    else:
+        raise ValueError('Invalid `side` parameter, pass either `left` or '
+                         '`right`.')
+
+    # get corresponding data
+    frequencies = frequencies[mask]
+    spectrogram = spectrogram[mask, ...]
```

Comment:
I think this needs to be changed for multi-dimensional signals:

If `signal.cdim == 1` this works, but if for example `signal.cdim == 2` the frequencies are in the third dimension. Is this related to pull requests for maintaining the cshape?

Reply:
The [..., mask, :] thing works for cdim > 1 :)

Reply:
And should also work for cdim == 1 after merging #666, right?

Reply:
yes, that should work